src/main/java/org/radarcns/phone/PhoneSensorManager.java

```diff
@@ -177,7 +246,56 @@ public void processLight(SensorEvent event) {
         send(lightTable, new PhoneLight(time, timeReceived, lightValue));
     }
 
-    public void processBatteryStatus(Intent intent) {
+    private void processGyroscope(SensorEvent event) {
+        // Not normalized axis of rotation in rad/s
+        float axisX = event.values[0];
+        float axisY = event.values[1];
+        float axisZ = event.values[2];
+
+        double timeReceived = System.currentTimeMillis() / 1_000d;
+
+        // nanoseconds uptime to seconds utc
+        double time = ( timeReceived - (System.nanoTime() - event.timestamp) / 1_000_000_000d );
+        send(gyroscopeTable, new PhoneGyroscope(time, timeReceived, axisX, axisY, axisZ));
+    }
+
+    private void processMagneticField(SensorEvent event) {
+        // Magnetic field in microTesla
+        float axisX = event.values[0];
+        float axisY = event.values[1];
+        float axisZ = event.values[2];
+
+        double timeReceived = System.currentTimeMillis() / 1_000d;
+
+        // nanoseconds uptime to seconds utc
+        double time = ( timeReceived - (System.nanoTime() - event.timestamp) / 1_000_000_000d );
+        send(magneticFieldTable, new PhoneMagneticField(time, timeReceived, axisX, axisY, axisZ));
+    }
+
+    private void processStep(SensorEvent event) {
+        // Number of step since listening or since reboot
+        float stepCount = event.values[0];
```

Comment:
could convert to `int` here: as per <https://developer.android.com/reference/android/hardware/Sensor.html#TYPE_STEP_COUNTER>, this will return a float with the fractional part set to zero. Likewise, lastStepCount could be an `int` with default to `-1` instead of `null`.